src/index.js

```diff
@@ -59,25 +76,97 @@ export class Wizard extends Component {
   }
 
   render () {
+    const { children } = this.props
+
     return (
       <Provider value={this.state}>
-        {this.props.children}
+        {children}
       </Provider>
     )
   }
+
+
+
+
+
+  /***************************************************************************\
+    Getters
+  \***************************************************************************/
+
+  get steps () {
+    const { children } = this.props
+
+    return React.Children.map(children, ({ props: { id } }, index) => id || index)
+  }
 }
 
 
 
 
 
-export const Step = ({ stepId, children }) => (
-  <Consumer>
-    {({ step, ...childProps }) => {
-      if (step === stepId) {
-        return children(childProps)
+export class Step extends Component {
+  /***************************************************************************\
+    Local Properties
+  \***************************************************************************/
+
+  static defaultProps = {
+    render: null,
+  }
+
+  static propTypes = {
+    id: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+    render: PropTypes.func,
+  }
+
+
+
+
+
+  /***************************************************************************\
+    Private Methods
+  \***************************************************************************/
+
+  _renderChildren (childProps) {
+    const {
+      children,
+      render,
+    } = this.props
+
+    const renderChildProps = { ...childProps }
+    delete renderChildProps.setStep
+    delete renderChildProps.wizardState
+
+    return React.Children.map(children, child => {
+      if (typeof child === 'function') {
+        return child(childProps)
+      }
+
+      if (render) {
+        return render(childProps)
       }
-      return null
-    }}
-  </Consumer>
-)
\ No newline at end of file
+
+      return React.cloneElement(child, renderChildProps)
+    })
+  }
+
+
+
+
+
+  /***************************************************************************\
+    Public Methods
+  \***************************************************************************/
+  render () {
+    const { id } = this.props
+
+    return (
+      <Consumer>
+        {({ currentStep, ...childProps }) => (
+          <div hidden={currentStep !== id}>
```

Comment:
I'm still not the biggest fan of preemptively applying this. We're still functionally mounting a component earlier than a user of the library would expect, there are potential performance problems on initial mount, and there is no benefit until we actually get time slicing.

Reply:
Your face is no benefit.